### ISSUE TITLE
Feature Request: Strategy for generating all of the `nn.Module` builtin layers

### ISSUE THREAD

**Issue by qthequartermasterman** [qthequartermasterman/hypothesis-torch#12]
There are many `torch.nn.Module` defined in the docs: https://pytorch.org/docs/stable/nn.html

We should support (or already support):
- [ ] [Containers](https://pytorch.org/docs/stable/nn.html#containers)
- [ ] [Convolution Layers](https://pytorch.org/docs/stable/nn.html#convolution-layers)
- [ ] [Pooling layers](https://pytorch.org/docs/stable/nn.html#pooling-layers)
- [ ] [Padding Layers](https://pytorch.org/docs/stable/nn.html#padding-layers)
- [ X ] [Non-linear Activations (weighted sum, nonlinearity)](https://pytorch.org/docs/stable/nn.html#non-linear-activations-weighted-sum-nonlinearity)
- [ X ] [Non-linear Activations (other)](https://pytorch.org/docs/stable/nn.html#non-linear-activations-other)
- [ ] [Normalization Layers](https://pytorch.org/docs/stable/nn.html#normalization-layers)
- [ ] [Recurrent Layers](https://pytorch.org/docs/stable/nn.html#recurrent-layers)
- [ ] [Transformer Layers](https://pytorch.org/docs/stable/nn.html#transformer-layers)
    - Although we do not yet support native Transformer layers, we do support a strategy that generates arbitrary huggingface Transformers models by parsing their corresponding config objects.
- [ X ] [Linear Layers](https://pytorch.org/docs/stable/nn.html#linear-layers)
- [ ] [Dropout Layers](https://pytorch.org/docs/stable/nn.html#dropout-layers)
- [ ] [Sparse Layers](https://pytorch.org/docs/stable/nn.html#sparse-layers)
- [ ] [Distance Functions](https://pytorch.org/docs/stable/nn.html#distance-functions)
- [ ] [Loss Functions](https://pytorch.org/docs/stable/nn.html#loss-functions)
- [ ] [Vision Layers](https://pytorch.org/docs/stable/nn.html#vision-layers)
- [ ] [Shuffle Layers](https://pytorch.org/docs/stable/nn.html#shuffle-layers)
- [ ] [DataParallel Layers (multi-GPU, distributed)](https://pytorch.org/docs/stable/nn.html#module-torch.nn.parallel)
- [ ] [Utilities](https://pytorch.org/docs/stable/nn.html#module-torch.nn.utils)
- [ ] [Quantized Functions](https://pytorch.org/docs/stable/nn.html#quantized-functions)
- [ ] [Lazy Modules Initialization](https://pytorch.org/docs/stable/nn.html#lazy-modules-initialization)
